### PR TITLE
Improving examples captured using Fiddler and changing the type for t…

### PR DESCRIPTION
**Pull request by gucalder**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.
…wo parameters to fix a serialization/deserialization bug

### CODE REVIEW

File: arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json [hunk 1]
```diff
@@ -64,21 +64,32 @@
...
-                  "Action"
-                ],
-                "retentionPolicy": {
-                  "enabled": true,
-                  "days": 1
-                }
+                "value": [
+                  {
+                    "id":"/subscriptions/df602c9c-7aa0-407d-a6fb-eb20c8bd1192/providers/microsoft.insights/logprofiles/default",
+                    "type":null,
```
Comment (amarzavery) [inline arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json]: is the service actually sending null? This will fail data validation.

**Comment by amarzavery** [comment]
@gucalder - Found multiple issues in the spec and also in examples
1. Invalid "arm-insights\2015-07-01\swagger\insightsManagementClient_DiagnosticsSettings.json" spec.
both the operations get & put have an unnecessary parameter subscriptionId. 

I pulled the spec from the AuxMon fork, removed those unnecessary parameters and ran the validation tool again. I found issues in the example. They are as follows, please take a look:

```
D:\sdk\openapi-validation-tools>node validate.js example D:\sdk\upstream\azure-rest-api-specs\arm-insights\compositeInsightsManagementClient.json

> Validating "examples" and "x-ms-examples" in D:\sdk\upstream\azure-rest-api-specs\arm-insights/2015-04-01/swagger/insightsManagementClient_Autoscale.json:


> No Errors were found.

> Validating "examples" and "x-ms-examples" in D:\sdk\upstream\azure-rest-api-specs\arm-insights/2015-07-01/swagger/insightsManagementClient_DiagnosticsSettings.json:


        > Operation: ServiceDiagnosticSettings_Get

                > example-in-spec
                        > RESPONSE_BODY_VALIDATION_ERROR: Found errors in validating the example for response with statusCode "200" in operation "ServiceDiagnosticSettings_Get".
                                > InnerErrors:
                                  1. code: INVALID_TYPE
                                     message: Expected type string but found type null
                                     path: location
                                     description: Resource location
                                  2. code: INVALID_TYPE
                                     message: Expected type string but found type null
                                     path: type
                                     description: Azure resource type
                                  3. code: INVALID_TYPE
                                     message: Expected type string but found type null
                                     path: properties,workspaceId
                                     description: The workspace ID (resource ID of a Log Analytics workspace) for a Log Analytics workspace to which you would like to send Diagnostic Logs. Example: /subscriptions/4b9e8510-67ab-4e9a-95a9-e2f1e570ea9c/resourceGroups/insights-integration/providers/Microsoft.OperationalInsights/workspaces/viruela2

        > Operation: ServiceDiagnosticSettings_CreateOrUpdate

                > example-in-spec
                        > RESPONSE_BODY_VALIDATION_ERROR: Found errors in validating the example for response with statusCode "200" in operation "ServiceDiagnosticSettings_CreateOrUpdate".
                                > InnerErrors:
                                  1. code: INVALID_TYPE
                                     message: Expected type string but found type null
                                     path: location
                                     description: Resource location
                                  2. code: INVALID_TYPE
                                     message: Expected type string but found type null
                                     path: type
                                     description: Azure resource type
                                  3. code: INVALID_TYPE
                                     message: Expected type string but found type null
                                     path: properties,workspaceId
                                     description: The workspace ID (resource ID of a Log Analytics workspace) for a Log Analytics workspace to which you would like to send Diagnostic Logs. Example: /subscriptions/4b9e8510-67ab-4e9a-95a9-e2f1e570ea9c/resourceGroups/insights-integration/providers/Microsoft.OperationalInsights/workspaces/viruela2

> Validating "examples" and "x-ms-examples" in D:\sdk\upstream\azure-rest-api-specs\arm-insights/2016-03-01/swagger/insightsManagementClient_AlertRules.json:


> No Errors were found.

> Validating "examples" and "x-ms-examples" in D:\sdk\upstream\azure-rest-api-specs\arm-insights/2016-03-01/swagger/insightsManagementClient_AlertRulesIncidents.json:


> No Errors were found.

> Validating "examples" and "x-ms-examples" in D:\sdk\upstream\azure-rest-api-specs\arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json:


        > Operation: LogProfiles_Get

                > example-in-spec
                        > RESPONSE_BODY_VALIDATION_ERROR: Found errors in validating the example for response with statusCode "200" in operation "LogProfiles_Get".
                                > InnerErrors:
                                  1. code: OBJECT_MISSING_REQUIRED_PROPERTY
                                     message: Missing required property: location
                                     path:
                                     description: An azure resource object
                                  2. code: OBJECT_MISSING_REQUIRED_PROPERTY
                                     message: Missing required property: properties
                                     path:
                                     description: The log profile resource.

        > Operation: LogProfiles_CreateOrUpdate

                > example-in-spec
                        > RESPONSE_BODY_VALIDATION_ERROR: Found errors in validating the example for response with statusCode "200" in operation "LogProfiles_CreateOrUpdate".
                                > InnerErrors:
                                  1. code: INVALID_TYPE
                                     message: Expected type string but found type null
                                     path: type
                                     description: Azure resource type
                                  2. code: INVALID_TYPE
                                     message: Expected type string but found type null
                                     path: properties,serviceBusRuleId
                                     description: The service bus rule ID of the service bus namespace in which you would like to have Event Hubs created for streaming the Activity Log. The rule ID is of the format: '{service bus resource ID}/authorizationrules/{key name}'.

> Validation Complete.
 
D:\sdk\openapi-validation-tools>
```

/cc @johanste @veronicagg @salameer

**Comment by amarzavery** [comment]
@gucalder - Which property "types" were changed?

File: arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json (hunk 1 above)
Comment (gucalder) [inline arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json]: Yes, that's a capture from the actual response from the service.

**Comment by gucalder** [comment]
Before, I was sending LogProfileProperties and ServiceDiagnosticSettings in the bodies of the PUT operations. It looked good in the spec, but the calls return a "missing properties field" (or similar) and fail.
Now I am sending LogProfileResource and ServiceDiagnosticSettingsResource and the calls work fine.

**Comment by gucalder** [comment]
Changing that behavior will need modifications in the services that are out of my control for the moment, i.e. that's Vinicius area.

File: arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json (hunk 1 above)
Comment (gucalder) [inline arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json]: I cannot tell why the service responds with null here and in the location. This is Vinicius area.  
Here I am just reflecting the actual behavior of the services.

Comment (gucalder) [inline arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json]: I can remove the examples if they cause issues in the validation tools. Or I can put there valid values for the tools to succeed - these are examples after all and I know the calls work as they currently are - and file a task for Vinicius to fix the behavior or change the spec and the examples if needed.
NOTE: after this week, I will start handing off the specs to the people that own the services. They have more knowledge and reasons for why the services work as they do.

**Comment by gucalder** [comment]
Checks are passing now. This is intended to fix this issue: https://github.com/Azure/azure-rest-api-specs/issues/887 too.

**Comment by gucalder** [comment]
This PR seems to be in the limbo. I am closing it and creating a new one.

**Pull request closed without merging**